### PR TITLE
refactor(bot): Refactor to use embeds templates

### PR DESCRIPTION
**Pull request by Swiizyy**

### CODE REVIEW

**Review by Swiizyy — COMMENTED** [review]
Test

File: .editorconfig [hunk 1]
```diff
@@ -4,11 +4,22 @@
...
 charset = utf-8
-trim_trailing_whitespace = false
+end_of_line = crlf
 insert_final_newline = true
+trim_trailing_whitespace = false
 
+[*.{js,ts}]
+indent_size = 4
+indent_style = tab
+block_comment_start = /*
```
Comment (nikolaischunk) [inline .editorconfig:15]: for what this required? what is it doing?

File: src/lib/utils/embed.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	BOT_NAME,
+	FAIL,
+	IS_CUSTOM_BOT,
+	SUCCESS,
+} from '../../helpers/provide/environment';
+
+type UniversalMessageOptions = Omit<BaseMessageOptions, 'flags'>;
+type UniversalInteractionOptions = Omit<InteractionReplyOptions, 'flags'>;
+
+export function generateEmbedWithDefault(embed: APIEmbed): APIEmbed {
```
Comment (nikolaischunk) [inline src/lib/utils/embed.ts:15]: rename to `generateDefaultEmbed`
```suggestion
export function generateDefaultEmbed(embed: APIEmbed): APIEmbed {
```

File: src/preconditions/IsPremium.ts [hunk 3]
```diff
@@ -5,19 +5,25 @@ export class UserPrecondition extends AllFlowsPrecondition {
...
 
 	public override async messageRun(message: Message) {
-		return this.premiumCheck(message.guildId!);
+		return this.premiumCheck(message.guildId);
 	}
 
-	private async premiumCheck(guild_id: string) {
-		const is_premium: boolean = (await this.container.utilities.guild.get.GuildById(guild_id))?.premium ?? false;
-		return is_premium ? this.ok() : this.error({ message: this.#message });
+	private premiumCheck(guildId: CommandInteraction['guildId']) {
```
Comment (nikolaischunk) [inline src/preconditions/IsPremium.ts:19]: maybe "outsource" this function to the prisma lib util?

Comment (Swiizyy) [reply src/preconditions/IsPremium.ts:19]: The Prisma utility will be out of date in a few days

**Review by nikolaischunk — CHANGES_REQUESTED** [review]

File: src/commands/general/birthday/remove.ts [hunk 4]
```diff
@@ -19,66 +16,44 @@ import thinking from '../../../lib/discord/thinking';
 )
 export class ListCommand extends Command {
 	public override async chatInputRun(interaction: Command.ChatInputInteraction<'cached'>) {
-		await thinking(interaction);
```
[the remark this answers is not in the document]
Comment (Swiizyy) [reply src/commands/general/birthday/remove.ts:22]: We don't need it here

File: src/commands/general/config/announcementChannel.ts [hunk 5]
```diff
@@ -26,45 +24,36 @@ export class AnnouncementChannelCommand extends Command {
...
-			});
+			return reply(
+				interaction,
+				interactionProblem(` I don't have permission to send messages in ${channelMention(channel.id)}.`),
+			);
 		}
 
-		const result = await Result.fromAsync(() =>
-			this.container.prisma.guild.update({
+		return this.container.prisma.guild
```
[the remark this answers is not in the document]
Comment (Swiizyy) [reply src/commands/general/config/announcementChannel.ts:33]: I intend to improve all this during the big change in the database

File: src/lib/utils/embed.ts [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+export function messageValidate(message: string): UniversalMessageOptions {
+	return {
+		content: '',
+		embeds: [validate(message)],
+		components: [],
+	};
+}
+
+export function interactionValidate(message: string, ephemeral = true): UniversalInteractionOptions {
```
[the remark this answers is not in the document]
Comment (Swiizyy) [reply src/lib/utils/embed.ts:51]: Yes, I'll do it :)

**Review by nikolaischunk — APPROVED** [review]

**Comment by Swiizyy** [comment]
LGTM 🚀

**Pull request merged**